### DISCUSSION TITLE
How to get the app's current version/build number programatically?

### DISCUSSION THREAD

**Discussion by JGKle** [AvaloniaUI/Avalonia#13682] [General]
On Desktop platforms, this works:

```
var assemblyName = Process.GetCurrentProcess().MainModule.FileName;
FileVersionInfo fvi = FileVersionInfo.GetVersionInfo(assemblyName);
AppVersion = "v" + fvi.ProductVersion;
```

But on Android, it does not. I found this SO post about how to get Android's version/build numbers in Xamarin: https://stackoverflow.com/questions/39659273/get-current-apps-build-version-xamarin-android

But couldn't find the equivalent in Avalonia?

Thanks in advance

Edit: to clarify, in the case of Android, it would be the version number & version name from AndroidManifest.xml

**Comment by maxkatz6** [comment]
> But couldn't find the equivalent in Avalonia?

Avalonia is a GUI framework.
Search for an equivalent in .NET Core. Or simply access Android APIs directly.

You can call the same `Application.Context.ApplicationContext.PackageManager.GetPackageInfo` method. But this API will be accessible from the project with "net7.0-android" TargetFramework (i.e. Android specific project).

**Reply by JGKle** [reply]
Thanks for the reply.

So as you described, I can get the version with:

```
var context = Application?.ApplicationContext;
var ver = context.PackageManager?.GetPackageInfo(context.PackageName, 0).VersionName;
```

But only in the AppName.Android project (aka the platform-specific project). The issue is I need to get to it in the Core project. I've spent ages Googling & haven't been able to figure it out, as every solution posted is specific to either MAUI or Xamarin, the other 2 GUI frameworks (in Xamarin, it's `Xamarin.Essentials.DeviceInfo`; in MAUI, it's `Microsoft.MAUI.ApplicationModel`). All the generic .NET Core stuff I've found returns version info from AssemblyInfo.cs, which seems to be Desktop specific - not from AndroidManifest.xml

Given that I can't seem to find a way to get the info directly in the Core project, is there some way to pass information in from the Android-specific project when the Core project is initialized?
